src/test/java/web/tests/FlipkartTest.java

```diff
@@ -1,34 +1,34 @@
 package web.tests;
 
 import com.codeborne.selenide.*;
+import com.training.base.BaseTest;
 import org.junit.jupiter.api.Assertions;
 import org.junit.jupiter.api.BeforeEach;
-import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.CsvFileSource;
 import web.pages.flipkart.HomePage;
 import web.pages.flipkart.SearchPage;
 
-public class FlipkartTest {
+public class FlipkartTest extends BaseTest {
     @BeforeEach
-    public void testSetUp(){
-        Configuration.timeout=5000;
-        Selenide.open("https://www.flipkart.com");
+    public void setUp() {
+        Selenide.open(map.get("url"));
     }
 
     /**
-     *getTestDetails :In this method shoes selection operation performed by using low to high filter
+     * testGetDetails :In this method shoes selection operation performed by using low to high filter
      */
-    @Test
-    public void testGetDetails()  {
-        String product ="shoes";
-        String filter="Price -- Low to High";
-        int pagelimit=2;
-        SearchPage searchPage=new HomePage().popUpCancel().setShoes(product).searchShoes().
-                 sortShoes(filter);
-               for(int page=1;page<=pagelimit;page++){
-                   if(page!=1){
-                      searchPage.selectPageNumber(page);
-                   }
-                   Assertions.assertTrue(searchPage.countingPagePrices(),"prices not matching");
-               }
+    @ParameterizedTest
+    @CsvFileSource(resources = "/Flipkart.csv")
+    public void testGetDetails(String product, String pagelimit) {
+        int max_page_limit= Integer.parseInt(pagelimit);
+        SearchPage searchPage = new HomePage().popUpCancel().setShoes(product).searchShoes().
+                sortShoes("Price -- Low to High");
+        for (int page = 1; page <= max_page_limit; page++) {
```

Comment:
Lets use Integer.parseInt(pageLimit) directly here, instead of defining a new variable.